src/plugins/ui_actions/public/cell_actions/components/cell_actions.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo, useRef } from 'react';
+import type { ActionExecutionContext } from '../../actions';
+import { InlineActions } from './inline_actions';
+import { HoverActionsPopover } from './hover_actions_popover';
+
+export interface CellActionField {
+  /**
+   * Field name.
+   * Example: 'host.name'
+   */
+  name: string;
+  /**
+   * Field type.
+   * Example: 'keyword'
+   */
+  type: string;
+  /**
+   * Field value.
+   * Example: 'My-Laptop'
+   */
+  value: string;
+}
+
+export interface CellActionExecutionContext extends ActionExecutionContext {
+  /**
+   * Ref to a DOM node where the action can add custom HTML.
+   */
+  extraContentNodeRef: React.MutableRefObject<HTMLDivElement | null>;
+
+  /**
+   * Ref to the node where the cell action are rendered.
+   */
+  nodeRef: React.MutableRefObject<HTMLDivElement | null>;
+
+  /**
+   * Extra configurations for actions.
+   */
+  metadata?: Record<string, unknown>;
+
+  field: CellActionField;
+}
+
+export enum CellActionsMode {
+  HOVER_POPOVER = 'hover-popover',
+  ALWAYS_VISIBLE = 'always-visible',
+}
+
+export interface CellActionsProps {
+  /**
+   * Common set of properties used by most actions.
+   */
+  field: CellActionField;
+  /**
+   * The trigger in which the actions are registered.
+   */
+  triggerId: string;
+  /**
+   * UI configuration. Possible options are `HOVER_POPOVER` and `ALWAYS_VISIBLE`.
+   *
+   * `HOVER_POPOVER` shows the actions when the children component is hovered.
+   *
+   * `ALWAYS_VISIBLE` always shows the actions.
+   */
+  mode: CellActionsMode;
+  showTooltip?: boolean;
+  /**
+   * It shows 'more actions' button when the number of actions is bigger than this parameter.
+   */
+  showMoreActionsFrom?: number;
+  /**
+   * Custom set of properties used by some actions.
+   * An action might require a specific set of metadata properties to render.
+   * This data is sent directly to actions.
+   */
+  metadata?: Record<string, unknown>;
+}
+
+export const CellActions: React.FC<CellActionsProps> = ({
+  field,
+  triggerId,
+  children,
```

Comment:
What is the purpose for the `children` property? Trying to find the description in `CellActionsProps`.
Probably we just should name it different, something like `triggeringElement` or so.

Reply:
The `children` property will be used for rendering the field content. `CellActions` wraps the cell content and provides extra functionality.

The `children` property is required for the hovering mode. `CellActions` wraps the children and displays a popover on hover. 
Here is an example: 
```  
<CellActions mode={CellActionsMode.HOVER_POPOVER} triggerId={TRIGGER_ID} field={FIELD}>
    Hover me
</CellActions>
 ```
 
 For inline mode, it isn't strictly required. But if it is provided, `CellActions` will display the action side-by-side with `children`.

We could consider exporting two components, one for hover and one for inline actions. But, in the first interaction, I decided to keep the API as simple as possible by exporting one component (We can easily change that).

Given that hover actions need to wrap the field value to work, `children` is the only property that allows the "wrapping" JSX syntax. 
```
<WrappingComponent>
  <WrappedComponent />
</WrappingComponent>
```
It is also built into typescript. So if a component is typed with `React.FC<>`, it already has a property named `children`.